packages/core/src/model/choropleth.ts

```diff
@@ -0,0 +1,80 @@
+// Internal Imports
+import { ChartModel } from './model';
+import { Tools } from '../tools';
+import { getColorScale } from '../services';
+
+/**
+ * Base thematic maps chart model layer
+ */
+export class ChoroplethModel extends ChartModel {
+	private _colorScale: any = undefined;
+
+	// Holds a mapping of geometry objects to data objects
+	// Allows us to access data faster
+	private _matrix = {};
+
+	constructor(services: any) {
+		super(services);
+	}
+
+	/**
+	 * @override
+	 * @param value
+	 * @returns string
+	 */
+	getFillColor(value: number) {
+		return this._colorScale(value);
+	}
+
+	/**
+	 * Helper function that will generate a dictionary
+	 */
+	getCombinedData() {
+		if (Tools.isEmpty(this._matrix)) {
+			const options = this.getOptions();
+			const data = this.getDisplayData();
+			if (
+				!Tools.isEmpty(data) &&
+				!Tools.isEmpty(options.geoData.objects.countries)
+			) {
+				/**
+				 * @todo
+				 * We can either use name or id by default to generate this dictionary
+				 * Curently id & name are standard in geoJSON. Unfortunately, topojson does not have any standard
+				 * so feature objects can have any key. We suggest that they include name or id at the very least
```

Comment:
would geoJSON be the better option then? Would we only support that or both?

Reply:
Just trying to have some context

Reply:
Going to create a new thread for this in this PR so we don't have to toggle resolved conversations in future.